libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+        if not retrieval_query:
+            retrieval_query = (
+                f"RETURN reduce(str='', k IN {text_node_properties} |"
+                " str + '\\n' + k + ': ' + coalesce(node[k], '')) AS text, "
+                "node {.*, `"
+                + embedding_node_property
+                + "`: Null, id: Null, "
+                + ", ".join([f"`{prop}`: Null" for prop in text_node_properties])
+                + "} AS metadata, score"
+            )
+
+        store = cls(
+            embedding=embedding,
+            search_type=search_type,
+            retrieval_query=retrieval_query,
+            node_label=node_label,
+            embedding_node_property=embedding_node_property,
+            **kwargs,
+        )
+
+        embedding_dimension, index_type, entity_label, entity_property = store.retrieve_existing_index()
+
+        # Raise error if relationship index type
+        if index_type == "RELATIONSHIP":
+            raise ValueError(
+                "`from_existing_graph` method does not support "
+                " existing relationship vector index. "
+                "Please use `from_existing_relationship_index` method"
+            )
+
+        # If the vector index doesn't exist yet
+        if not index_type:
+            print(f"Making index for node label: {node_label}")
+            store.create_new_index(node_label=node_label)
+        # If the index already exists, check if embedding dimensions match
+        elif (
+            embedding_dimension and not store.embedding_dimension == embedding_dimension
+        ):
+            raise ValueError(
+                f"Index on Node {store.node_label} already exists."
+                "The provided embedding function and vector index "
+                "dimensions do not match.\n"
+                f"Embedding function dimension: {store.embedding_dimension}\n"
+                f"Vector index dimension: {embedding_dimension}"
+            )
+        # FTS index for Hybrid search
+        if search_type == SearchType.HYBRID:
+            fts_node_label = store.retrieve_existing_fts_index()
+            # If the FTS index doesn't exist yet
+            if not fts_node_label:
+                store.create_new_keyword_index(text_node_properties)
+            else:  # Validate that FTS and Vector index use the same information
+                if not fts_node_label == store.node_label:
+                    raise ValueError(
+                        "Vector and keyword index don't index the same node label"
+                    )
+
+        # Populate embeddings
+        
+        while True:
+            fetch_query = (
+                f"MATCH (n:`{node_label}`) "
+                f"WHERE n.`{embedding_node_property}` IS null "
+                "AND any(k IN $props WHERE n[k] IS NOT null) "
+                "RETURN id(n) AS id, "
+                "reduce(str='', k IN $props | str + '\\n' + k + ': ' + coalesce(n[k], '')) AS text "
```

Comment:
is that the right way to generate the text from which embeddings are going to be created? shouldn't we just get the `text` attribute ?

Reply:
Changed to 
```
fetch_query = (
                f"MATCH (n:`{node_label}`) "
                f"WHERE n.`{embedding_node_property}` IS null "
                "AND any(k IN $props WHERE n[k] IS NOT null) "
                "RETURN id(n) AS id, "
                "coalesce(n.text, '') AS text "
                "LIMIT 1000"
            )
```

Reply:
I change it to `coalesce(n.text, '') AS text`